# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/HTMLEditor.h
# Commit: 555d2ca16477
# Full Hash: 555d2ca1647762dfcdb7651a260d133782782ba7
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-09-21 09:52:11
# Regressor Bug: 1789967
# File Overlap Count: 1
# Description:
#   Bug 1789967 - part 1: Make `TextEditor` and `HTMLEditor` implement `EditorBase::InitEditorContentAndSelection` by themselves r=m_kato CLOSED TREE
#   
#   The method is enough simple, and uses bad cast from point of view of OOP.
#   Therefore, this patch make the sub classes implement the method only for each.
#   
# ==============================================================================

diff -r fb7ca98a6881 -r 555d2ca16477 editor/libeditor/HTMLEditor.h
--- a/editor/libeditor/HTMLEditor.h	Wed Sep 21 04:40:46 2022 +0300
+++ b/editor/libeditor/HTMLEditor.h	Wed Sep 21 00:20:26 2022 +0000
@@ -2682,9 +2682,6 @@
    * InitEditorContentAndSelection() may insert `<br>` elements and padding
    * `<br>` elements if they are required for `<body>` or document element.
    * And collapse selection at the end if there is no selection ranges.
-   * XXX I think that this should work with active editing host unless
-   *     all over the document is ediable (i.e., in design mode or `<body>`
-   *     or `<html>` has `contenteditable` attribute).
    */
   [[nodiscard]] MOZ_CAN_RUN_SCRIPT nsresult InitEditorContentAndSelection();
 